apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestAudioActivity.java

```diff
@@ -133,6 +133,21 @@ public void onReceive(Context context, Intent intent) {
         }
     };
 
+    private BroadcastReceiver mScreenStateReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getAction().equals(Intent.ACTION_SCREEN_OFF)) {
+                if (mStreamSniffer != null) {
+                    mStreamSniffer.stopStreamSniffer();
+                }
+            } else if (intent.getAction().equals(Intent.ACTION_SCREEN_ON)) {
+                if (mStreamSniffer != null && mAudioState < AUDIO_STATE_CLOSING) {
```

Comment:
Should the audio state compare in this way?

Reply:
I think it's fine because closeAudio looks as follows:

~~~
    // Make synchronized so we don't close from two streams at the same time.
    public synchronized void closeAudio() {
        if (mAudioState >= AUDIO_STATE_CLOSING) {
            Log.d(TAG, "closeAudio() already closing");
            return;
        }
~~~

Reply:
That is also something I want us to fix. There is not any guarantee this will always work.

Reply:
Thanks! Done!